Tests/MongoSwiftSyncTests/SyncTestUtils.swift

```diff
@@ -46,6 +46,7 @@ extension MongoSwiftTestCase {
             try database.collection(collName).drop()
             collection = try database.createCollection(collName, options: options)
         }
+        try collection.deleteMany([:])
```

Comment:
I would think this is unnecessary given that we either just create the collection, or just dropped and recreated it. can you explain why this was added?

Reply:
I was trying to figure out how to fix this behavior of collection drop only dropping from primaries and sharded replica sets. Although this fix seemed to break everything instead which I also don't understand why because the collection should hold nothing at this point. 

https://docs.mongodb.com/manual/reference/method/db.collection.drop/

Reply:
So the issue is that in sharded clusters, each mongos retains an internal cache of the available collections, and apparently re-creating a dropped collection may retain some of the previous one's documents (at least on that mongos?). This deleteMany ensures those documents are actually cleared out correctly. The linked documentation in @bynn's comment includes a small tidbit about having to do this, but doesn't really explain why its necessary.

This is important for this test in particular because the `setUp` inserts a few documents which need to be cleared out.

Reply:
It's probably worth including a comment inline here explaining this for future readers.

Reply:
done